exporter/prometheusremotewriteexporter/compliance/compliance_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compliance
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/compliance/remote_write/cases"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/processor/batchprocessor"
+	"go.opentelemetry.io/collector/service"
+	"go.opentelemetry.io/collector/service/parserprovider"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver"
+)
+
+const configFormat = `
+receivers:
+  prometheus:
+    config:
+      scrape_configs:
+        - job_name: 'test'
+          scrape_interval: 1s
+          static_configs:
+            - targets: [ '%s' ]
+processors:
+  batch:
+exporters:
+  prometheusremotewrite:
+    endpoint: '%s'
+service:
+  pipelines:
+    metrics:
+      receivers: [prometheus]
+      processors: [batch]
+      exporters: [prometheusremotewrite]
+`
+
+var (
+	logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	tests  = []func() cases.Test{
```

Comment:
I'm not sure that duplicating the test harness here is the best approach.  Can we instead have an action that runs on PR that pulls the compliance test repo and runs the tests there against a build artifact?